pkg/kubectl/cmd/util/factory.go

```diff
@@ -165,6 +166,24 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			if cfg.GroupVersion != nil {
 				cmdApiVersion = *cfg.GroupVersion
 			}
+			client, err := clients.ClientForVersion(&unversioned.GroupVersion{Version: "v1"})
+			if err != nil {
+				fmt.Printf("Failed to create client: %v", err)
+				return nil, nil
+			}
+			list, err := client.ThirdPartyResources(api.NamespaceDefault).List(api.ListOptions{})
+			if err != nil {
+				fmt.Printf("Failed to populate third party resources: %v", err)
+				return nil, nil
+			}
+			gvs, _, err := thirdpartyresourcedata.ExtractGroupVersionKind(list)
+			if err != nil {
+				fmt.Printf("Failed to get third party group/versions")
+				return nil, nil
+			}
+			registered.SetThirdPartyAPIGroupVersions(gvs)
```

Comment:
I'm a little wary of setting a global variable as a side-effect of calling a factory method.  This doesn't seem like an ideal location to do this.